### ISSUE TITLE
The weekly and monthly events calendars do not maintain the dates you are viewing when switching between the list and grid versions.

### ISSUE THREAD

**Issue by mmafe** [ccswbs/hjckrrh#825]
The weekly and monthly events calendars do not maintain the dates you are viewing when switching between the list and grid versions. In other words, if I was on the grid view, looking at a week that was three weeks ahead, and I switched to the list view, I would now be back on the current week.

# Who does it affect?
All anonymous users

# What does it affect?
Affects both the weekly and monthly calendars.

# Why does it matter?
If viewers are navigating through these calendars and click forward a number of times to find the date they are looking for, it can be frustrating to have to start from scratch because you changed your display mode. 

I feel like this issue would be twice as frustrating if you were doing this with assistive technology.  For example, if you're using a screen-reader and someone sent you a link to the grid version for events that are 3 weeks ahead, you'd have to:
- switch to the list version for it to be accessible
- realize that you are now 3 weeks behind (which you can only find from the sub-heading since it's not in the page title)
- find the next button (which you don't know is there until you navigate through the page to get to it)
- click the next button 3 times (and each time you click it, remember the page has been refreshed, so you have to navigate from the top of the pagedown to the next button again)
...at which point, you can finally read the list you wanted to review.

If the list is meant to be the accessible alternative for the grid, you should be able to get to the list view for the grid of dates that you are looking at, otherwise I don't think it's a reasonable alternative. We also cannot guarantee that users will start out looking at the list (e.g. if someone sends them a link to the grid view).

# Steps to Reproduce
1. Go to /events/week or /events/month/grid
2. Press the next button one or more times, so the week is in the future and has content. Take note of the week (e.g. Week of October 8, 2017)
3. Press View Calendar as a List. The week is now set to the current week instead of the week you were looking at.
